massdriver-application-azure-vm-docker/network.tf

```diff
@@ -0,0 +1,16 @@
+locals {
+  zone_name                = var.dns.enable_dns ? regex(".*/dns[z|Z]ones/(.*)$", var.dns.zone_id)[0] : null
+  zone_resource_group_name = var.dns.enable_dns ? regex(".*/resource[g|G]roups/(.*)/providers", var.dns.zone_id)[0] : null
+}
+
+data "azurerm_virtual_network" "main" {
+  name                = local.vnet_name
+  resource_group_name = local.vnet_resource_group
+}
+
+data "azurerm_subnet" "default" {
+  # does it have to be called default?
+  name                 = "default"
+  virtual_network_name = data.azurerm_virtual_network.main.name
+  resource_group_name  = data.azurerm_virtual_network.main.resource_group_name
+}
```

Comment:
I don't think these are needed, because we're creating a subnet for the VMSS in `subnet.tf`, and using auto-cidr to configure it. 

What this seems to be doing is using the `default` subnet coming from the VNet bundle and using it in `ip_configurations` block within VMSS. I think what we want is to use the subnet that gets created in this deployment.

Reply:
Where is the subnet for the vm being created?

Reply:
Ahh I know that file  `subnet.tf`. We use it for the endpoint, we might be able to use it for the VM too.